### ISSUE TITLE
[@typescript-eslint/explicit-function-return-type] context not recognized

### ISSUE THREAD

**Issue by NonPolynomial** [typescript-eslint/typescript-eslint#2267]
**Repro**

```YAML
rules:
  '@typescript-eslint/explicit-function-return-type': error
```

```TS
const list: readonly {
  readonly foo: (bar: string) => boolean // set return type
}[] = [
  {
    foo: (bar) => true, // <- ESLint complains here
  },
]
```

**Expected Result**
ESLint not complaining about missing return type within the array

**Actual Result**
ESLint complaining about missing return type within the array

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.34.0` |
| `@typescript-eslint/parser`        | `2.34.0` |
| `TypeScript`                       | `3.9.3` |
| `ESLint`                           | `7.1.0` |
| `node`                             | `14.2.0` |
| `npm`                              | `6.14.4` |

**Comment by bradzacher** [comment]
Same issue as #2183

**Comment by dman777** [comment]
I am hitting this issue also with 

```
  rules: {
    '@typescript-eslint/explicit-function-return-type': 'off',
  },
```

**Comment by dman777** [comment]
bump...this needs to get fixed, please

**Comment by bradzacher** [comment]
This is an open source project run by volunteers.

If something is important to you, feel free to submit a PR.

If you are unable to submit a PR - that's perfectly fine! But please don't bump an issue thread with a bump comment. These comments just spam people with notifications. It doesn't do anything good and just annoys people, reducing the likelihood of someone else fixing this.

**Comment by dman777** [comment]
The work around of using a older version no longer exists. It's dependencies are no longer available:

```
npm ERR! Could not resolve dependency:
npm ERR! peer eslint@"^5.0.0 || ^6.0.0" from @typescript-eslint/parser@2.34.0
npm ERR! node_modules/@typescript-eslint/parser
npm ERR!   dev @typescript-eslint/parser@"^2.30.0" from the root project
```

I will never submit a PR because I do not use typescript in my own code. I only use it when I have to and in this case other peoples projects. 

If types in javaScript were truly beneficial than this bug would not exist.

**Comment by NonPolynomial** [comment]
seems to be fixed with #2805 
🙌🎉
